docs/cli/orchestration/index.md

```diff
@@ -32,3 +32,75 @@ The default order of execution can be altered in a stack's configuration. For de
+
+These two ways to orchestrate commands in Terramate give you everything you need to manage your commands, from simple tasks to bigger, more complicated workflows. Whether you're running one-off commands or managing a whole series of actions, Terramate has you covered.
+
+
+## Change Detection Integration
+
+Terramate offers seamless integration with [change detection](../change-detection/index.md), allowing users to optimize command orchestration by focusing only on the modified stacks. This section highlights how users can leverage the `terramate run` and `terramate script run` commands in conjunction with the `--changed` flag to facilitate this process.
+
+By pairing the `--changed` flag with the `terramate run` command, users can instruct Terramate to execute commands exclusively on stacks that have undergone modifications since the last execution. Similarly, when using the `terramate script run` command, appending the `--changed` flag enables users to trigger the execution of Terramate Scripts solely on stacks with detected changes.
+
+This feature makes it easier for users to handle resources effectively, especially when there are changes in the setup of their infrastructure.
+
+## Sequential and Parallel Command Orchestration
+Terramate CLI supports both sequential and parallel execution of commands to orchestrate stacks, thus accommodating both dependent and independent stacks. The CLI utilizes a [fork-join model](https://en.wikipedia.org/wiki/Fork%E2%80%93join_model) to execute the sequential parts (dependent stacks requiring specific order) and parallel parts (independent stacks that can be executed in any order). By leveraging this approach, Terramate ensures efficient execution while maintaining accuracy and consistency across deployments.
+
+### Sequential
+Stacks that have a dependency on other stacks need to be run sequentially.
+Terramate cli runs [nested stacks](../stacks/nesting.md) in sequence as per the order of execution defined in the stack's configuration.
+
+In Terramate, by default, commands are executed sequentially. When adhering to the default order of execution for a stack hierarchy as illustrated below:
+```sh
+/vpc
+  /network_acl
+  /internet_gateway
+  /subnet
+    /network_interface
+    /ec2
+  /route_table
+  /security_group
+```
+The sequence of execution follows this pattern:
+
+1. `vpc`
+2. `network_acl`
+3. `internet_gateway`
+4. `subnet`
+   - `network_interface`
+   - `ec2`
+5. `route_table`
+6. `security_group`
+
+As an illustration, to execute a command sequentially across all stacks within a particular directory:
+```bash
+terramate run --chdir stacks/vpc -- terraform init
+```
+### Parallel
+
+Terramate facilitates parallel execution, enabling independent stacks to run in parallel, thereby offering significant time savings, particularly during commands like `terramate run terraform init`. Despite the parallel nature of execution, Terramate ensures that the order of execution is still respected.
```

Comment:
```suggestion
Terramate facilitates parallel execution, enabling independent stacks to run in parallel, thereby offering significant time savings, particularly during commands like `terraform init`. Despite the parallel nature of execution, Terramate ensures that the order of execution is still respected.
```
What do you think?

Reply:
I think it's right to mention `terraform init` command only as some users can run raw Terraform commands as well, and Terramate being an IAC agnostic tool supports parallel execution in such cases too.